### PR TITLE
Modified for RAMP use

### PR DESCRIPTION
**Pull request by lamlaaaam**
Changes in this PR:
1. Set the SLAM to use IMU data.
2. Enabled floor detection.
3. Created a node to call the "map_saver" service upon ctrl C (To be consistent with how Velodyne LOAM and LeGO-LOAM maps are being saved)

### CODE REVIEW

File: launch/hdl_graph_slam.launch [hunk 1]
```diff
@@ -16,7 +16,7 @@
   <arg name="robot_odom_frame_id" default="robot_odom" />
 
   <!-- transformation between lidar and base_link -->
-  <node pkg="tf" type="static_transform_publisher" name="lidar2base_publisher" args="0 0 0 0 0 0 base_link velodyne 10" />
+  <!--<node pkg="tf" type="static_transform_publisher" name="lidar2base_publisher" args="0 0 0 0 0 0 base_link velodyne 10" />-->
```
Comment (aarushg22) [inline launch/hdl_graph_slam.launch:19]: why is this commented? is it not required anymore?

File: src/hdl_graph_slam/map_saver.cpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+int main(int argc, char **argv)
+{
+    ros::init(argc, argv, "map_saver");
+    ros::NodeHandle n;
+
+    // Form message
+    float res = 0.05;
+    std::string des = argv[1];
```
Comment (aarushg22) [inline src/hdl_graph_slam/map_saver.cpp:24]: read these values from a rosparam.

**Review by aarushg22 — CHANGES_REQUESTED** [review]

**Comment by aarushg22** [comment]
Provide a brief description of the changes you did for this PR.

File: launch/hdl_graph_slam.launch (hunk 1 above)
Comment (lamlaaaam) [reply launch/hdl_graph_slam.launch:19]: If playing from bag file, the bag has the tfs. If real time, the collect.launch file has the tfs. So no need here.

File: src/hdl_graph_slam/map_saver.cpp (hunk 2 above)
Comment (lamlaaaam) [reply src/hdl_graph_slam/map_saver.cpp:24]: res as a rosparam sounds good, but I think destination should be a cmd line arg to be consistent with the other SLAMs. Actually why not just make res a cmd line arg as well with default 0.05, then expose as an arg in process.launch. e.g. `roslaunch 3d_Slam_tools process algo:=hdl res:=0.1 ...`

**Pull request closed without merging**